pkg/epp/scheduling/types/types.go

```diff
@@ -33,12 +33,23 @@ type LLMRequest struct {
 	Prompt string
 	// Headers is a map of the request headers.
 	Headers map[string]string
+
+	// filterMetadata is a map of metadata in the request
+	metadata map[string]any
```

Comment:
I don't see a point in private field when there are Get/Set public functions.
private field is usually used if we want to limit the field to be read only from outside of the package and then we define only a Get function.

additionally godoc still specifies `filterMetadata`, need to fix the typo